src/ShelfLayoutManager.Infrastructure/Ingestion/SkuFileParser.cs

```diff
@@ -0,0 +1,96 @@
+using FluentResults;
+using ShelfLayoutManager.Core;
+
+namespace ShelfLayoutManager.Infrastructure.Ingestion;
+
+/// <inheritdoc>
+public class SkuFileParser : ISkuFileParser
+{
+    public async Task<Result<IList<Sku>>> Parse(Stream stream, CancellationToken ct)
+    {
+        List<Sku> skus = [];
+
+        using StreamReader reader = new(stream);
+
+        int i = 0;
+        string? line;
+        while ((line = await reader.ReadLineAsync(ct)) != null)
```

Comment:
Nit: Could have used a csv parsing library, maybe.
But not going to fault anyone for handwriting a csv parser real quickly.